### PR TITLE
Fix ServiceWorker implementation

### LINKED ISSUE #3195: ServiceWorker bug bash
Here's what currently happens on the second visit to Spectrum:

screen shot 2018-05-30 at 10 43 46

As you can see, the ServiceWorker serves our app shell HTML for images that aren't cached! 🤦‍♂️  I'm working on fixing this, and also the still-occurring ServiceWorker update error from #3011

### PR DESCRIPTION
**Pull request by mxstbr**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

This fixes the ServiceWorker implementation by upgrading to the new version of the `offline-plugin` and using its `appShell` option.

Deploy with these changes: https://spectrum-pzcshptgvd.now.sh Note how the images on the homepage are correctly served even on second visit when the app is cached :tada:

Fixes #3195

### CODE REVIEW

**Comment by mxstbr** [comment]
This might also fix the ServiceWorker updating error we've been seeing, although I'm not sure since I don't know how to consistently repro in prod and thusly also don't know how to verify that it doesn't happen anymore after this patch.

File: hyperion/index.js [hunk 1]
```diff
@@ -118,7 +118,16 @@ try {
   console.error(err);
 }
 app.use(
-  express.static(path.resolve(__dirname, '..', 'build'), { index: false })
+  express.static(path.resolve(__dirname, '..', 'build'), {
+    index: false,
+    setHeaders: (res, path) => {
+      // Don't cache the serviceworker in the browser
+      if (path.indexOf('sw.js')) {
+        res.setHeader('Cache-Control', 'no-store');
```
Comment (ryota-murakami) [inline hyperion/index.js:126]: i think `no-store` is the best choice🙌💯 (rather than `no-cache` etc)

**Comment by brianlovin** [comment]
Uhhh that CI failure is odd: https://circleci.com/gh/withspectrum/spectrum/7473?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link - no version of cypress installed?

Going to guess that's nothing to do with this branch :P

File: package.json [hunk 2]
```diff
@@ -71,6 +70,7 @@
     "cors": "^2.8.3",
     "cryptr": "^3.0.0",
     "css.escape": "^1.5.1",
+    "cypress": "^3.0.0",
```
Comment (brianlovin) [inline package.json]: Oh wait - this breaks CI

**Review by brianlovin — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (mxstbr) [reply package.json]: Oopsie that was accidental, reverting.

Comment (ryota-murakami) [reply package.json]: - This is off topic

oh, cypress 3.0.0 available now👀
https://twitter.com/oskarhane/status/1001807076384038912

change log
https://github.com/cypress-io/cypress/commit/353190d94ec90a0d9e0995d86eb50a35df85b523

**Review by brianlovin — APPROVED** [review]

**Pull request merged**